src/sentry/metrics/minimetrics.py

```diff
@@ -25,17 +25,22 @@ def patch_sentry_sdk():
     real_add = MetricsAggregator.add
     real_emit = MetricsAggregator._emit
 
-    @wraps(real_add)
-    def tracked_add(self, ty, *args, **kwargs):
-        real_add(self, ty, *args, **kwargs)
+    @metrics_noop
```

Comment:
I would honestly fix it in a different way, since this code works under the assumption that the SDK sets internally the thread local. I would honestly implement it differently and more reliably by just adding the `@enter_minimetrics` decorator to the monkeypatched functions and protecting the call to the SDK inside of the `MinimetricsMetricsBackend`.

Reply:
What do you think?

Reply:
Similar to https://github.com/getsentry/sentry/pull/57782 but with the `finally` implementation since it doesn't clear up the state in case of an exception.

Reply:
> since this code works under the assumption that the SDK sets internally the thread local

I don't believe the code makes such an assumption as `metrics_noop` sets the therad local.

The risk of a second thread local is this one:

> I temporarily considered moving metrics_noop to sentry itself, but the protection still makes sense in the main SDK. Reason being that the transport is in parts (via client) invoked from the flush logic and this can create metrics cycles even without the sentry specific metrics system.

I believe if they do not use the same thread local we have paths where only minimetrics protects itself, but not the outer code path.

Reply:
Ah ok so it sets also the thread local, I thought it was only checking it and not triggering the function.